packages/react-reconciler/src/ReactFiberLazyComponent.js

```diff
@@ -50,9 +50,9 @@ export function readLazyComponentType<T>(lazyComponent: LazyComponent<T>): T {
       thenable.then(
         moduleObject => {
           if (lazyComponent._status === Pending) {
-            const defaultExport = moduleObject.default;
+            const _result = moduleObject ? moduleObject.default : moduleObject;
             if (__DEV__) {
-              if (defaultExport === undefined) {
+              if (result === undefined) {
```

Comment:
```suggestion
              if (_result === undefined) {
```
`result` is from the upper scope.